src/inmanta/protocol/methods_v2.py

```diff
@@ -508,3 +508,51 @@ def resource_did_dependency_change(
     :raises BadRequest: When this endpoint in called while the resource with the given resource version is not
                         in the deploying state.
     """
+
+
+@typedmethod(path="/resource", operation="GET", arg_options=methods.ENV_OPTS, client_types=[ClientType.api], api_version=2)
+def resource_list(
+    tid: uuid.UUID,
+    limit: Optional[int] = None,
+    first_id: Optional[model.ResourceVersionIdStr] = None,
```

Comment:
Why `ResourceVersionIdStr` and not `ResourceIdStr`

Reply:
Because of the unique constraints